internal/controller/replication.storage/finalizers.go

```diff
@@ -65,43 +66,44 @@ func (r *VolumeReplicationReconciler) removeFinalizerFromVR(logger logr.Logger,
 	return nil
 }
 
-// addFinalizerToPVC adds the VR finalizer on the PersistentVolumeClaim.
-func (r *VolumeReplicationReconciler) addFinalizerToPVC(logger logr.Logger, pvc *corev1.PersistentVolumeClaim) error {
-	if !slices.Contains(pvc.ObjectMeta.Finalizers, pvcReplicationFinalizer) {
-		logger.Info("adding finalizer to PersistentVolumeClaim object", "Finalizer", pvcReplicationFinalizer)
-		pvc.ObjectMeta.Finalizers = append(pvc.ObjectMeta.Finalizers, pvcReplicationFinalizer)
-		if err := r.Client.Update(context.TODO(), pvc); err != nil {
+// addFinalizerToPVC adds the VR, VGR finalizer on the PersistentVolumeClaim.
+func addFinalizerToPVC(client client.Client, logger logr.Logger, pvc *corev1.PersistentVolumeClaim,
```

Comment:
What is the reason to remove this function (and others) from the struct, and make it a stand-alone helper? Callers should not need to care about the `client.Client` that is passed.

Reply:
We are now referencing this function in multiple controllers. So, as to mkae the `client` value independent of the caller I moved it out of the struct. Otherwise, we would be just duplicating the function in both the controllers.